### ISSUE TITLE
Direct tag edit

### ISSUE THREAD

**Issue by Aury88** [openstreetmap-ng/openstreetmap-ng#90]
Hi. thank you for this wonderful project! 
Something [I always feel missing](https://www.openstreetmap.org/user/Aury88/diary/35165) in classic osm.org is a way to directly add/edit/remove tags.
when we select an object on the map we can see all tags applied to it in the sidebar but we can't modify them directly (we can only search their meaning on osmWiki).
![immagine](https://github.com/user-attachments/assets/b9b4ce5b-fc28-4a07-8ce0-38cb97b30848)

 We have to load an application re-select the element  or to copy its ID, past it in [Level0](http://level0.osmz.ru/) and then finally apply the desired changes. I personally find this very strange for a website made for the mappers.

I would like and find very usefull that, if the user is logged in and there is an object selected, to have an "Edit tags" button over the tags table and, if edited, a "Discarge" and a "Save" buttons on the bottom of a changeset comment text-box

I know this is could be difficult to implement but i think this would be a very appreciated feature (clearly after the feature parity) from all the contributors that only need to make small/single changes to tag.

Regards.

**Comment by Zaczero** [comment]
I like that idea! Simple tag editor for simple use cases. It sounds like it will be quite simple to implement.

**Comment by starsep** [comment]
There is browser extension https://github.com/Zverik/osmtags-editor which implements that for the current website. Personally I find it extremely useful for small edits (I'm not a fan of iD), which I would otherwise do in Level0.

**Comment by Aury88** [comment]
@starsep thank you very much! did not know that. 
I consider the existence of a that extension a proof of interest in this feature and the fact that I and part of the italian community did know of that extension as a proof of lack of communication of available tools ...if osm-NG will be able to suggest tools/applications in a convenient way this project will have solved one of the most serious problems (imho) of the osm project. thank you very much again.

**Comment by KonradowyPL** [comment]
> if osm-NG will be able to suggest tools/applications in a convenient way this project will have solved one of the most serious problems (imho) of the osm project. 

osm-NG will have software recommendation list (#24)

**Comment by Zaczero** [comment]
> > if osm-NG will be able to suggest tools/applications in a convenient way this project will have solved one of the most serious problems (imho) of the osm project.
> 
> osm-NG will have software recommendation list (#24)

But we will also have this feature, it sounds like a core feature of the site :slightly_smiling_face:

**Comment by Zaczero** [comment]
The goal is to:

- When logged in, display "Edit text" link-like button (btn btn-link) in the same line as "Tags" title
- When clicked, hide the tags and show a \<textarea>, inside a \<form>, hide the "Edit text" button
- \<form> has Discard and Submit buttons at the bottom
- \<form> is inside a Jinja2 template, not created by JavaScript
- Tags contain the raw key/values in some data-* html attribute
- data-* html attribute contains JSON-encoded dictionary of tags, so there is just one data field per all tags group
- After edit text button click, fill the \<textarea> with the tags in form of `key=value` (read from data-* html attribute
- On submit click, let the browser submit a POST request as normal
- On discard click, hide the \<form> and show the original tags list and the "Edit text" button

We will want to hook it up to the new API 0.7 so currently backend implementation should be skipped. Let's focus on the frontend now.

**Comment by sabas** [comment]
It could be done also in the iD way with autocomplete 
![image](https://github.com/user-attachments/assets/9060a3ab-b322-4981-8f29-6bdb34e96188)

**Comment by Aury88** [comment]
> It could be done also in the iD way with autocomplete 

Yeah. in this way  it reduce typos and at the same time it keep a consistent style between osm and iD application. That's exactly what I was thinking.

**Comment by Zaczero** [comment]
I have created a [new issue](https://github.com/openstreetmap-ng/openstreetmap-ng/issues/94) which will deal with taginfo integration after the ground work is finished first (this issue).

**Comment by goteusz-maszyk** [comment]
The only concern is that it would create tons of micro-changesets when you want to modify e.g. multiple nodes or ways. Could be solved by adding changes to a changeset and then finally applying everything with a separate page. (Kinda like streetcomplete) Obviously newcomers would have to be notified about this.

**Comment by Aury88** [comment]
Imho this feature is designed to cover a specific need,  a single tag  edit, so in the gap between the note edit and the use of editing tools . for more complex/large editing  there are already suitable tools designed for this and way better at this ( ie integrated with Q/A check etc) . the impediment of saving changesets with >1 changes prevented his features from being easily used to make large changes and this is imho a good thing.

**Comment by RSimmons2021** [comment]
/attempt #90

I’ll implement the frontend-only slice from the maintainer spec: logged-in/latest-element Edit text control, server-rendered form, JS toggle/prefill from JSON tags data, and no backend mutation until API 0.7.

**Comment by nexicturbo** [comment]
Submitted PR #247 for this $20 bounty: https://github.com/openstreetmap-ng/openstreetmap-ng/pull/247

It implements the requested frontend-only direct tag edit slice: logged-in/latest-version visibility, JSON-backed tag prefill, editable key=value textarea, required comment field, and normal POST form wiring while leaving backend API 0.7 mutation out of scope as requested.

**Comment by ikoomm** [comment]
@Zaczero I checked #90 against current main (`0f27e65`). The sidebar now renders through `app/views/index/element.tsx`, and the element RPC service exposes reads only. The linked specification still requests a Jinja form with normal POST submission and no backend implementation.

Would a Preact editor matching the current sidebar be preferred, and which endpoint should its Submit action target? I can cover latest/visible-element gating, raw `key=value` prefill, required comment, discard/reset, and focused UI checks for this $20 issue. This question is about wiring the requested frontend scope correctly; I have not submitted another PR yet.
